int64 start_x
int64 start_y
int64 end_x
int64 end_y
bool get_extra_queen
bool capture_piece
bool castle_right
bool castle_left
bool enpassent
---
int64 error_code
